app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -303,6 +361,7 @@ private void hideInfoLayoutArcPointer() {
 
     @Override
     public void onDestroy() {
+        timer.cancel();
         super.onDestroy();
```

Comment:
This reminds me: `super.onDestroy` should probably be last, it's often a safer default (no matter what Android docs tell)—I've dug some discussions about it, can find it again.